lib/interface.py

```diff
@@ -241,7 +237,12 @@ def get_socket(self):
         return s
 
     def run(self):
-        socket = self.get_socket()
+        try:
+            socket = self.get_socket()
+        except OSError:
+            traceback.print_exc()
+            socket = None
+
```

Comment:
As discussed in #1096, the exception handler now covers more code, but catches a more specific exception class.

Reply:
Hmm.  I didn't think this is what you meant.  I'm pretty sure this will break things in the desktop app.  Look at the code within the previous BaseException coverage -- some of it intentionally would raise AssertionErrors and other stuff like that.

I'm pretty sure this will break Qt and iOS... I need to review this more thoroughly. It really looks like the "normal flow" of this code was to catch all exceptions (as ugly as that is)..

Reply:
Or.. I could be wrong.  Will review. :)

Reply:
The only code covered by the previous `BaseException` handler was these two lines:

https://github.com/Electron-Cash/Electron-Cash/blob/7cc851b9aa5054741ebde3b66d107706bad9cd15/lib/interface.py#L196-L198

Is there any reason why that would throw an exception that's not a subclass of `OSError`?

Reply:
Jesus christ you're right.  I misread the formatting of the `except BaseException` somehow and saw it visually as being outside the current indentation level (and thus I wrongly assumed it caught exceptions raised inside the `ssl.SSLError` handler, etc).

100% my bad. Sorry!

You're right -- I think your change is fine.